receiver.go

```diff
@@ -805,7 +815,7 @@ func (r *Receiver) muxReceive(fr frames.PerformTransfer) error {
 	}
 	debug.Log(1, "deliveryID %d before push to receiver - deliveryCount : %d - linkCredit: %d, len(messages): %d, len(inflight): %d", r.msg.deliveryID, r.l.deliveryCount, r.l.availableCredit, len(r.messages), r.inFlight.len())
 	// send to receiver
-	if receiverSettleModeValue(r.l.receiverSettleMode) == ReceiverSettleModeSecond {
+	if !r.msg.settled {
```

Comment:
Even more awesome than the link check. This is perfect for Event Hubs (and receive and delete in SB).